server/src/formatting.ts

```diff
@@ -3,56 +3,76 @@ import {
     DocumentFormattingParams,
     TextEdit,
     DocumentRangeFormattingParams,
+    Position,
+    Range,
     WorkspaceFolder
 } from 'vscode-languageserver/node';
 import {
     TextDocument
 } from 'vscode-languageserver-textdocument';
 import { PerlDocument, PerlElem, NavigatorSettings } from "./types";
-import {  nLog } from "./utils";
+import {  async_execFile, getPerlimportsProfile, nLog } from "./utils";
 import { dirname, join } from 'path';
+import Uri from 'vscode-uri';
 import { execFileSync } from 'child_process';
 import { getPerlAssetsPath } from "./assets";
 
 
 export function formatDoc(params: DocumentFormattingParams, txtDoc: TextDocument, settings: NavigatorSettings): TextEdit[] | undefined {
+    const start = Position.create( 0,0 );
+    const end = Position.create( txtDoc.lineCount, 0 );
+    const range = {start, end} as Range;
```

Comment:
You should be able to use `Range.create` (source [here](https://github.com/microsoft/vscode-languageserver-node/blob/main/types/src/main.ts#L203)) and avoid the type cast.

`const range = Range.create( Position.create( 0,0 ), Position.create( txtDoc.lineCount, 0 ) );`

Also, doesn't look like the `text` variable is being used in this function.

With that, you could probably replace the whole body of the function with just a `return` statement:

```ts
return maybeReturnEdits(
  Range.create( Position.create( 0,0 ), Position.create( txtDoc.lineCount, 0 ) ),
  txtDoc,
  settings
);
```
That said, you could certainly keep the `range` variable if you find that more readable